src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -95,8 +95,42 @@ export class TechRecordSummaryComponent implements OnInit {
     this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
   }
 
+  processAxleEvent(axleEvent: any): void {
```

Comment:
```suggestion
  processAxleEvent(axleEvent: any): void {
      if (axleEvent.axles.length < this.vehicleTechRecordCalculated.axles.length) {
          this.vehicleTechRecordCalculated = cloneDeep(this.vehicleTechRecordCalculated);
          let axelToRemove = findAxelToRemove(axelEvent);
          this.vehicleTechRecordCalculated.axles = this.vehicleTechRecordCalculated.axles
          .filter(ax => ax.axleNumber !== axleToRemove)
          .forEach((ax: Axle) => {
               if (ax.axleNumber! > axleToRemove) {
               ax.axleNumber! -= 1;
               }
         });
  }
  
  findAxelToRemove(axleEvent: any): integer {
  let previousAxleRow = 1;
      axleEvent.axles.every((ax: Axle) => {
        if (ax.axleNumber === previousAxleRow) {
          previousAxleRow += 1;
        } else {
          return ax.axleNumber! - 1;
        }
        return ax.axleNumber + 1;
      });
  }
  
```

Reply:
Something like this, where the functionality can also be extracted would increase readability and understanding of the codebase here.

Secondly here I think the main goal of this function is to see if there was an axel removed or added and then do something based on that. So we can abstract more out and have a better name then processAxelEvent